app/test_main.py

```diff
@@ -2,0 +1,18 @@
+from _pytest.monkeypatch import MonkeyPatch
+import app.main
+
+
+def test_valid_google_url(monkeypatch: MonkeyPatch) -> None:
+    monkeypatch.setattr("app.main.valid_google_url", lambda url: True)
+
+
+def test_has_internet(monkeypatch: MonkeyPatch) -> None:
```

Comment:
Consider renaming the test function to something more descriptive, such as `test_internet_connection_mock`, to better reflect its purpose and align with the checklist's recommendation for descriptive test names.